src/main/java/cl/transbank/webpay/example/controller/webpay/WebpayPlusController.java

```diff
@@ -71,17 +71,23 @@ public ModelAndView create(HttpServletRequest request) {
     public ModelAndView commit(HttpServletRequest request) {
         String tokenWs = request.getParameter("token_ws");
         Map<String, Object> details = new HashMap<>();
-
-        if (tokenWs == null) {
+        String buyOrder = (String) request.getSession().getAttribute("buyOrder");
+        String sessionId = (String) request.getSession().getAttribute("sessionId");
+        if(request.getParameter("TBK_TOKEN")==null && tokenWs==null || request.getParameter("TBK_TOKEN").isEmpty() && tokenWs.isEmpty())
```

Comment:
Suggestion: The logic inside the if should be moved to a specific variable, such as transactionHasTimeout or something similar.

Reply:
The suggested changes were made